File: Player.java
import java.util.Vector;

public class Player {

    int hp = 100;
    int currentDamage = 1;

    double x_pos;
    double y_pos;
    double z_pos;

    Camera camera;

    void move(int direction, int distance){
        System.out.println("Moving in " + direction + ":" + distance);
    }

    void shoot(){
        //Shoot hitscan middle of camera
    }

};

File: player.cpp
#include <iostream>
#include "player.h"

        void Player::move(int direction, int distance){
            std::cout << "Moving in " << direction << ":" << distance << std::endl;
        }

        void Player::shoot(){
            //Shoot hitscan middle of camera
        }

        void Player::setXPos(double x_pos){
            this->x_pos = x_pos;
        }

        double Player::getXPos() {
            return x_pos;
        }